### DISCUSSION TITLE
Creating a custom `dl` html element with cva gives a type-error in IDE and gives a `Error: Maximum call stack size exceeded` when app is ran

### DISCUSSION THREAD

**Discussion by yaikohi** [joe-bell/cva#46] [Q&A] [answered]
**Describe the bug**
Using the syntax from the docs ("React with tailwindcss") I tried to make it work for a `<dl>` element but am getting type errors on the return type and a call stack error when running the application.

**To Reproduce**

1.  Copy and paste the following code in vscode / IDE with ts server:
```tsx
const styledDescriptionList = cva("dl", {
    variants: {
        intent: {
            primary: [
                "grid",
                "grid-cols-2",
                "mx-2",
                "gap-x-8",
                "text-ctp-subtext0"
            ]
        }
    }
})

export interface StyledDescriptionListProps
    extends React.HTMLAttributes<HTMLDListElement>,
        VariantProps<typeof styledDescriptionList> {}

export const StyledDescriptionList: React.FC<StyledDescriptionListProps> = ({
    className,
    intent,
    ...props
}) => (
    <dl
        className={StyledDescriptionList({ intent, class: className })}
        {...props}
    />
)
```
3. See error:
`
(property) class: string | undefined
Argument of type '{ intent: "primary" | null | undefined; class: string | undefined; }' is not assignable to parameter of type 'StyledDescriptionListProps'.
  Object literal may only specify known properties, and 'class' does not exist in type 'StyledDescriptionListProps'.ts(2345)
No quick fixes available
` in the IDE 
and `InternalError: too much recursion` or `Error: Maximum call stack size exceeded` on the browser (firefox developer edition).

**Expected behavior**
There should not be a type error and I don't think there should be a call stack error either. 

**Screenshots**
![image](https://user-images.githubusercontent.com/94071837/196782999-95945b78-4996-4db7-9e9c-0062b8b98a30.png)

**Additional context**
My package.json:

```json
{
// ...
"dependencies": {
    "@prisma/client": "^4.4.0",
    "@remix-run/node": "^1.7.2",
    "@remix-run/react": "^1.7.2",
    "@remix-run/vercel": "^1.7.2",
    "@vercel/node": "^2.5.22",
    "bcryptjs": "^2.4.3",
    "cva": "npm:class-variance-authority@^0.2.4",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@catppuccin/tailwindcss": "^0.1.0",
    "@remix-run/dev": "^1.7.2",
    "@remix-run/eslint-config": "^1.7.2",
    "@remix-run/serve": "^1.7.2",
    "@types/bcryptjs": "^2.4.2",
    "@types/react": "^18.0.21",
    "@types/react-dom": "^18.0.6",
    "autoprefixer": "^10.4.12",
    "concurrently": "^7.4.0",
    "eslint": "^8.25.0",
    "postcss": "^8.4.17",
    "prisma": "^4.4.0",
    "tailwindcss": "^3.1.8",
    "typescript": "^4.8.4"
  },
  "engines": {
    "node": ">=14"
  }
// ...
}

```
I am using ubuntu 22.04, pnpm, remix and tailwindcss.
I also made custom UL and LI elements that didn't give me any type errors.

I am not very good with typescript so if this is not actually a bug... I am sorry. 
Let me know if I need to provide more context.

**EDIT**

My custom UL and LI elements don't actually work for me either although I don't get any typeErrors, the class names don't actually get updated when I run the application.
![image](https://user-images.githubusercontent.com/94071837/196788213-0d4ee090-46ff-4672-a214-a9ee236b3f26.png)

**Comment by joe-bell** [comment, marked as answer]
Hey! Can you share your tsconfig? Or even better a Stackblitz showing the error live?

**Reply by yaikohi** [reply]
Hello! 

I made a stackblitz: https://stackblitz.com/edit/node-gkrg3q?file=app/routes/index.tsx

It shows the same behavior

**Reply by joe-bell** [reply]
Thanks @yaikohi! I'm away from the computer for a few days, but will take a look as soon as I can

**Reply by yaikohi** [reply]
No worries! Thank you for the response

**Reply by joe-bell** [reply]
Fixed it for you! Easy one to miss:

You're self-referencing the component, not the `cva` function 😊 

```diff
export const StyledDescriptionList: React.FC<StyledDescriptionListProps> = ({
	className,
	intent,
	...props
}) => (
	<dl
-			className={StyledDescriptionList({ intent, class: className })}
+			className={styledDescriptionList({ intent, class: className })}
			{...props}
	/>
)
```

Also don't forget to [create a `tailwind.config.js` with `content`](https://tailwindcss.com/docs/content-configuration) defined to allow JIT to create your classes

**Reply by yaikohi** [reply]
... this is very embarrassing I am sorry for the inconvenience but also thankful - thank you
